Code/GraphMol/Descriptors/USRDescriptor.h

```diff
@@ -0,0 +1,130 @@
+//
+//  Copyright (c) 2013, Novartis Institutes for BioMedical Research Inc.
+//  All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Novartis Institutes for BioMedical Research Inc.
+//       nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written
+//       permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+/*! \file USRDescriptor.h
+
+  \brief Contains the USR descriptor. Use MolDescriptors.h in client code.
+
+*/
+#ifndef __RD_USR_H__
+#define __RD_USR_H__
+
+#include <Geometry/point.h>
+#include <Numerics/Vector.h>
+
+namespace RDKit {
+class ROMol;
+class Conformer;
+namespace Descriptors {
+/*!
+  Calculates the ultra-fast shape recognition (USR) descriptor
+
+  Reference: P. J. Ballester, W. G. Richards, JCC (2007), 28, 1711 - 1723.
+
+  Derived from RDKit Python implementation of Jan Domanski
+  who derived his code from Adrian Schreyer's code:
+  http://hg.adrianschreyer.eu/usrcat/src/70e075d93cd2?at=default
+
+  \param mol          the molecule of interest
+  \param descriptor   storage for the computed USR descriptor
+  \param confId       the conformer Id
+
+*/
+void USR(const ROMol &mol, std::vector<double> &descriptor, int confId = -1);
```

Comment:
Feel free to ignore and I'll merge.

I just think that these are some of the changes we would like to do for c++11, and since there are no performance penalties we might as well start with new code and keep the API the same.